kpi/signals.py

```diff
@@ -100,3 +100,11 @@ def post_delete_asset(sender, instance, **kwargs):
     # Remove all permissions associated with this object
     ObjectPermission.objects.filter_for_object(instance).delete()
     # No recalculation is necessary since children will also be deleted
+
+    # Update parent's languages if this object is a child of another asset.
+    try:
+        if instance.parent:
+            instance.parent.update_languages()
+    except Asset.DoesNotExist:  # `parent` may exists in DJANGO models cache but not in DB
+        pass
+
```

Comment:
idea to avoid accidentally catching an unexpected exception in `update_languages()`:
```
try:
    parent = instance.parent
except Asset.DoesNotExist:
    pass
else:
    parent.update_languages()
```

Reply:
(unless there's a legitimate reason why the exception would not be raised _until_ `update_languages()` runs)